core_lib/session/user_security.py

```diff
@@ -37,7 +40,12 @@ def token_to_session_object(self, token):
 
     def _secure_entry(self, request, policies):
         session_obj = None
-        if self.cookie_name in request.COOKIES:
+        if WebHelpersUtils.get_server_type() == WebHelpersUtils.ServerType.DJANGO and self.cookie_name in request.COOKIES:

```

Comment:
it's impossible to ask 
```
if Django AND CookieExists
   .... 
else: 
   get cookie as flask 
```

what if this is Django but the cookie is not set? 
then you assume this is a flask. while it is Django?

here is my suggestion: 
1. try it
2. make sure all tests are running
3. test this on real flask and real Django 

```
    def _secure_entry(self, request, policies):
        cookies = {}
        if WebHelpersUtils.get_server_type() == WebHelpersUtils.ServerType.DJANGO:
            cookies = request.COOKIES
        elif WebHelpersUtils.get_server_type() == WebHelpersUtils.ServerType.FLASK:
            cookies = request.cookies
        token = cookies.get(self.cookie_name)
        session_obj = self.from_session_data(self.token_handler.decode(token)) if token else None
        return self.secure_entry(request, session_obj, policies)
```

Thank you

Reply:
Actually, we have three conditions here, In the first condition if it is django but the cookie is not present in that case we will not assume this to be flask but we are sure that either the cookie is not present and we have to return unauthorized or it is flask and then in the second condition we will check that if the flask cookie is present and if the flask cookie is present then good otherwise we will be sure that both the cookies are not present and we will return unauthorized in the third condition